src/common/config.h

```diff
@@ -67,6 +68,7 @@ void setNeoMode(bool enable);
 void setUserName(const std::string& type);
 void setUpdateChannel(const std::string& type);
 void setSeparateUpdateEnabled(bool use);
+void setGameInstallDirs(const std::vector<std::filesystem::path>& settings_install_dirs_config);
```

Comment:
```suggestion
void setGameInstallDirs(std::span<const std::filesystem::path> settings_install_dirs_config);
```

Reply:
The variable used in this function 

void setGameInstallDirs(std::span<const std::filesystem::path> settings_install_dirs_config) {
    settings_install_dirs = settings_install_dirs_config;
}

can be passed to the addGameInstallDir function, which is written for vector (as well as a similar function for removing install folders)

config.cpp: L354:
bool addGameInstallDir(const std::filesystem::path& dir) {
    if (std::find(settings_install_dirs.begin(), settings_install_dirs.end(), dir) ==
        settings_install_dirs.end()) {
        settings_install_dirs.push_back(dir);
        return true;
    }
    return false;
} 

My understanding is that the vector would have to be stored so that we can freely add and move directories, so at some point the span would have to be converted to vector to use these functions. If this understanding is correct, perhaps we can just leave the function as vector?

Reply:
Looking at the function again I think indeed this should be made back into a vector as it copies the entries into the config

Reply:
ok will revert then